crates/vsock/src/main.rs

```diff
@@ -305,35 +306,35 @@ fn main() {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::fs::File;
     use std::io::Write;
+    use std::{fs::File, path::Path};
     use tempfile::tempdir;
 
     impl VsockArgs {
         fn from_args(
             guest_cid: u64,
-            socket: &str,
-            uds_path: &str,
+            socket: impl AsRef<Path>,
+            uds_path: impl AsRef<Path>,
             tx_buffer_size: u32,
             groups: &str,
         ) -> Self {
             VsockArgs {
                 param: Some(VsockParam {
                     guest_cid,
-                    socket: socket.to_string(),
-                    uds_path: uds_path.to_string(),
+                    socket: socket.as_ref().to_owned(),
+                    uds_path: uds_path.as_ref().to_owned(),
```

Comment:
I - personally - still think that `.into()` is simpler... But if nobody else complains, I will remain silent :P

Reply:
+1 for `into()` but I don't have a strong opinion.

Just to understand better, what is the downside of `into()`?
More cryptic/magical?

Reply:
It is unreadable. I prefer the T::from variant

Reply:
I usually prefer less verbose, but I understand that for reading it then becomes even worse than the C casts.

Again, I don't have a strong opinion on that. I agree with @Ablu, and I'd used `.into()` since I don't see much value on that verbosity, but I see your point, so no objection on my side.

Reply:
I mean you still need as_ref, it is more of a into/to_owned and that is pretty clear from the language semantic. Anyway, I am happy to change things however the project maintainers thinks is ok/better

Reply:
Ah, right, so IMHO go head with your approach ;-)

Reply:
> +1 for `into()` but I don't have a strong opinion.
> 
> Just to understand better, what is the downside of `into()`? More cryptic/magical?

It is mostly a "automatic" `T::from`. I also prefer `T::from` if the types are relevant. When the code is a "just make this that type, I do not care about the details" I am happy with `.into()`. In this case, one could also use `.to_path_buf()`.

Looking at the code again... This function will _always_ copy the arguments. So we could just make it take a PathBuf and give the caller the choice. Then callers can just move in a PathBuf that is no longer required elsewhere or do a copy if required. This allows us to drop the `.as_ref().to_owned()` in entirey.

Here, this function is used from a single place - a test. So I am not concerned about any performance issues. But I think using `PathBuf` would require the least cognitive complexity to reason about? :) It would just be a `.clone()` on the caller side (since it is needed later again).

But let's not haggle about one-off test code too much... I do not have strong feelings here and happy with any solution.

Reply:
Right, I agree with what you said.